### ISSUE TITLE
more on suspected waiter issue

### ISSUE THREAD

**Issue by nigoroll** [varnishcache/varnish-cache#2202]
Re #2179 #2117 and maybe others:
@bsdphk had asked me to update systems on which these issues surface to a recent master commit and I did update one system, but kept it on standby without production load after a crash due to unhandled SIGPIPE (which has not yet been understood properly either). But even without production load we logged the following crash (had to be manually re-formatted - sorry if that's imperfect):

```
Panic at: Wed, 01 Feb 2017 18:59:02 GMT
 Assert error in HTTP1_Session(), http1/cache_http1_fsm.c line 406:
 Condition(req-doclose 0) not true.
 errno = 131 (Connection reset by peer)
 thread = (cache-worker)
 version = varnish-trunk revision 6f9595e
 ident = -jsolaris,-smalloc,-smalloc,-hcritbit,ports
 now = 17659710.884808 (mono), 1485975541.724341 (real)
 Backtrace:
 4527d6: /usr/local/sbin/varnishdpan_ic+0x196 0x4527d6
 473b3d: /usr/local/sbin/varnishdHTTP1_Session+0xb4d 0x473b3d
 473cd9: /usr/local/sbin/varnishdhttp1_req+0x59 0x473cd9
 46aad2: /usr/local/sbin/varnishdWRK_Thread+0x4a2 0x46aad2
 46af53: /usr/local/sbin/varnishdpool_thread+0x33 0x46af53
 ffffbf7fff257daa: /lib/amd64/libc.so.1_thrp_setup+0x8a 0xffffbf7fff257daa
 ffffbf7fff2580c0: /lib/amd64/libc.so.1_lwp_start+0x0 0xffffbf7fff2580c0
 req = 2e86030 {
 vxid = 0,
 transport = HTTP/1 { state = HTTP1::NewReq }
 step = 0x0,
 req_body = R_BODY_INIT,
 restarts = 0,
 esi_level = 0,
 sp = 2dbd330 {
 fd = 23, 
vxid = 3294478,
 t_open = 1485975541.559822,
 t_idle = 1485975541.559822,
 transport = HTTP/1 { state = HTTP1::NewReq }
 client = 54.90.70.164 44914,
 },
 ws = 2e86218 { id = \req\, {s, f, r, e} = {2e8f018, +24, 0, +225264}, },
 http_conn = 2e86140 {
 fd = 23 (@2dbd348),
 doclose = NULL,
 ws = 2e86218 { Already dumped, see above },
 {rxbuf_b, rxbuf_e} = {2e8f018, 2e8f030},
 {pipeline_b, pipeline_e} = {0, 0},
 content_length = 0,
 body_status = none,
 first_byte_timeout = 0.000000,
 between_bytes_timeout = 0.000000,
 },
 http req = 2e862b0 { ws = 0 {

```

**Comment by bsdphk** [comment]
I fixed that one yesterday, 03602d9fb9e89379c648ad95df11bcc69dd1d004

**Comment by hermunn** [comment]
Backport review: This is H2, so not for 4.1.
